UpgradedVehicles/VehicleUpgrader.cs

```diff
@@ -2,13 +2,47 @@
 {
     using System;
     using System.Collections.Generic;
+    using System.Reflection;
     using Common;
+    using SMLHelper.V2.Handlers;
     using UnityEngine;
     using UpgradedVehicles.SaveData;
 
     public class VehicleUpgrader : MonoBehaviour
     {
-        internal static readonly ICollection<TechType> CommonUpgradeModules = new List<TechType>()
+#if BELOWZERO
+        private static Type SeaTruckSpeedManager => Type.GetType("SeaTruckSpeedUpgrades.SeaTruckSpeedManager,SeaTruckSpeedUpgrades", false, false);
+        
+        // Going for the private field rather than the public property, because the private field is static and the public property isn't
+        private static FieldInfo SpeedManagerAccelerationsInfo => SeaTruckSpeedManager?.GetField("accelerations", BindingFlags.NonPublic | BindingFlags.Static);
+        private static Dictionary<TechType, float> _speedManagerAccelerations;
+
+        private static Dictionary<TechType, float> SpeedManagerAccelerations
+        {
+            get
+            {
+                if (_speedManagerAccelerations == null)
+                {
+                    if (SeaTruckSpeedManager != null)
+                        _speedManagerAccelerations = (Dictionary<TechType, float>)(SpeedManagerAccelerationsInfo.GetValue(null));
+                }
+
+                return _speedManagerAccelerations;
+            }
+        }
+#endif
+
+        private static readonly Dictionary<TechType, float> VehicleEfficiencyBonuses = new Dictionary<TechType, float>()
+        {
+#if BELOWZERO
+            { TechType.SeaTruckUpgradeEnergyEfficiency, 1f },
+#endif
+            { TechType.VehiclePowerUpgradeModule, 1f }
+        };
+
+        private static readonly Dictionary<TechType, float> VehicleEfficiencyPenalties = new Dictionary<TechType, float>();
+
+        private static readonly HashSet<TechType> CommonUpgradeModules = new HashSet<TechType>()
```

Comment:
`CommonUpgradeModules` was originally intended to be the set of upgrades that could be equipped by both the Seamoth and the Prawn Suit. 

Given the changes, I'm not sure if that's what you had in mind.  
Can you confirm what you are using this set for?